### PR TITLE
Docker/Panel fixes and improvements

### PR DESCRIPTION
**Pull request by luiztauffer**
- [x] fix requirements for Dockerfile
- [x] improve docker/readme
- [x] github action to build and publish Image on new releases
- [x] local mount option, depends on #300

### CODE REVIEW

**Comment by bendichter** [comment]
lgtm! @CodyCBakerPhD could you take a look as well since you know more about this type of devops stuff?

**Comment by CodyCBakerPhD** [comment]
Yes, I've been keeping my eye on it and will make some comments later as I notice a couple things while adding similar stuff to https://github.com/catalystneuro/neuroconv/pull/383

**Comment by luiztauffer** [comment]
@CodyCBakerPhD any specific things you see that should be blocking this merge? I wanted to get the dockerized mode updated and start releasing the images alongside the pypi releases, it will make it very easy to run Panel remotely always with the latest updates

File: .github/workflows/auto-publish.yml
Comment (CodyCBakerPhD) [inline .github/workflows/auto-publish.yml:48]: Please use a `secrets.GITHUB_USERNAME`, the `github.actor` is whoever originally pushes the workflow, which may not be the same user the `GITHUB_TOKEN` credentials match

File: .github/workflows/auto-publish.yml
Comment (CodyCBakerPhD) [inline .github/workflows/auto-publish.yml:11]: Highly recommend setting this to `push` for a commit or 2 on this PR to test/ensure that it works before merging it to main, otherwise it's going to be a hassle to debug

Comment (CodyCBakerPhD) [reply .github/workflows/auto-publish.yml:11]: (and then remove `push` before merging, of course)

File: .github/workflows/auto-publish.yml
Comment (CodyCBakerPhD) [inline .github/workflows/auto-publish.yml:49]: Recommend using a static token that matches the specified username to keep them in sync - note that it will need package permissions as well

File: docs/source/contents/running_with_docker.md
Comment (CodyCBakerPhD) [inline docs/source/contents/running_with_docker.md:8]: I'd love to test this - can you either get this to trigger from this branch - or manually push your build [[instructions](https://github.com/catalystneuro/neuroconv/pull/383/files#diff-fcc22aa372714b7faa21a54039c3a7fbe1abd106c1f15f9f8e6cf85588dfe592R28-R47)]

File: docker/Dockerfile
Comment (CodyCBakerPhD) [inline docker/Dockerfile:1]:
```suggestion
FROM python:3.10
LABEL org.opencontainers.image.source=https://github.com/NeurodataWithoutBorders/nwbwidgets
```

The GHCR likes having these set

File: docker/Dockerfile_publish
Comment (CodyCBakerPhD) [inline docker/Dockerfile_publish:1]:
```suggestion
FROM python:3.10
LABEL org.opencontainers.image.source=https://github.com/NeurodataWithoutBorders/nwbwidgets
```

File: docker/Dockerfile
Comment (luiztauffer) [reply docker/Dockerfile:1]: @CodyCBakerPhD can you send me some reference for this? I'm interested in why that is the case, might use it in other places as well if it's important

File: .github/workflows/auto-publish.yml
Comment (luiztauffer) [reply .github/workflows/auto-publish.yml:11]: I agree, will do that

File: docker/Dockerfile
Comment (CodyCBakerPhD) [reply docker/Dockerfile:1]: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#labelling-container-images

Comment (CodyCBakerPhD) [reply docker/Dockerfile:1]: description and licenses also not a terrible idea I suppose but I haven't messed with that yet

Comment (luiztauffer) [reply docker/Dockerfile:1]: but setting these helped solve any problem, in specific? I still can't see a clear advantage of doing that
usually we can tag the same image with multiple targets, e.g. if we want to push to ECR, Dockerhub, Azure CR, etc... and in some cases that might be interesting to do. Why would we want to fix that label to ghcr?

Comment (CodyCBakerPhD) [reply docker/Dockerfile:1]: image

When I did it to my images, everything worked as expected. I believe I recall seeing errors otherwise

**Review by bendichter — DISMISSED** [review]

File: nwbwidgets/version.py
Comment (luiztauffer) [inline nwbwidgets/version.py:1]:
```suggestion
version = "0.11.3"
```

File: .github/workflows/auto-publish.yml
Comment (luiztauffer) [inline .github/workflows/auto-publish.yml:9]:
```suggestion
  push:
    branches:
      - panel-fixes-and-improvements
```

Comment (luiztauffer) [inline .github/workflows/auto-publish.yml:9]:
```suggestion
  push:
    branches:
      - panel-fixes-and-improvements
```

**Review by bendichter — APPROVED** [review]

**Pull request merged**
